### ISSUE TITLE
Canvas Color Settings

### ISSUE THREAD

**Issue by ShahNafis** [UNCG-DAISY/Coastal_Image_Labeler_Doodler#14]
Using this package [here](https://www.npmjs.com/package/react-canvas-draw) ([demo](https://embiem.github.io/react-canvas-draw/)) provides a nice way to draw on images. There are a few settings like the color of the brush and brush width.

Should the question set for an image have predefined colors (as in each color meanings something) or should the user select their own colors from a color wheel or something and assign meanings to those, or both with predefined + user-defined colors

@dbuscombe-usgs @ebgoldstein

**Comment by ddbuscombe** [comment]
Nice. That looks ideal. 

I vote predefined very prescribed colors, such as water is blue, sky is cyan, rocks are grey, etc. Partly because the annotations (when viewed as an image) and resulting segmentations are more perceptual, and partly because it is easier to deal with in the database (far fewer possible color-classname pair combinations to keep track of).

Wider point, I think it was my impression that this tool would serve only certain imagery and class sets that we define, am I right? If so, the user doesn't get to upload any data for their personal use, only contributing to a common good, i.e. a large dataset consistently labelled throughout. 

@ebgoldstein we should discuss what the plan is for the first use-case, I have only vague recollections - is it segmenting washover deposits in NOAA imagery?

**Comment by ebgoldstein** [comment]
working backwards — yes @dbuscombe-usgs , you are correct, the images would be served in the same way (loaded on a VM, we decide what people see, etc.)

as for colors:
> predefined very prescribed colors, such as water is blue, sky is cyan, rocks are grey, etc

So, to implement in the workflow we have:
- an admin would enable the painting tool for a given catalog of images
- in the Q set definition, the admin would define all the annotations (water, rock, etc.) and all the corresponding colors (blue, grey, etc)

As for the first catalog to label, i don't have a preference. If we use the NOAA images, then we already have loaded on the VM and its a bit easier..

@dbuscombe-usgs , if you are ok with NOAA, then do you have ideas re: class and color (IIRC, you have done segmenting work with these before?

**Comment by ddbuscombe** [comment]
Makes sense to me to go with NOAA - familiar, already loaded, follow-up to previous work, etc. I've segmented a few low-res images for water/nowater.

As for class/colors, maybe there is a strong argument for just one class (washover) and everything else is 0/null class. Beyond that, I would ask how much context is required and/or how much added value additional classes/complexity give. Others I can imagine being somewhat consistently annotated would include

* water
* non-washover sand and sand dune (probably toegther)
* washover sand on marsh
* washover sand on dune
* washover sand on road/developed/urban
* marsh
* ?

I would advocate for no 'other' type classes if we went multiclass. I'd also advocate for no 'high/middle/low' or 'high/low' style classes - too subjective
